python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -383,7 +383,21 @@ def _get_fast_num_partitions(self, current_time: Optional[datetime] = None) -> O
             return None
 
         if self.is_basic_daily:
-            return (last_partition_window.start - first_partition_window.start).days + 1
+            # work around DST issues with counting days by moving the windows to UTC first
+            return (
+                datetime(
```

Comment:
hm it's not 100% obvious to me why / that this method works -- I do basically believe that it does the job, but the concept of creating a new datetime object that is known to be different from the source objects (as it's in a different timezone) feels a bit shaky.

what would make a bit more sense to my mind would be something along the lines of:

```python
delta_microseconds = last_partition_window.start.timestamp() - first_partition_window.start.timestamp()
return datetime.timedelta(microseconds=delta_microseconds).days + 1
```

because in that world you're always working with things in the context of the correct time. accepting now and do with this suggestion what you wish (and it's possible that suggestion doesn't even work for some arcane reason)